pkg/internal/apis/certmanager/validation/certificaterequest.go

```diff
@@ -107,6 +113,80 @@ func ValidateCertificateRequestSpec(crSpec *cmapi.CertificateRequestSpec, fldPat
 	return el
 }
 
+// ValidateCertificateRequestApprovalCondition will ensure that only a single
+// 'Approved' or 'Denied' condition may exist, and that they are set to True.
+func ValidateCertificateRequestApprovalCondition(crConds []cmapi.CertificateRequestCondition, fldPath *field.Path) field.ErrorList {
+	var (
+		approvedConditions []cmapi.CertificateRequestCondition
+		deniedConditions   []cmapi.CertificateRequestCondition
+		el                 = field.ErrorList{}
+	)
+
+	for _, cond := range crConds {
+		if cond.Type == cmapi.CertificateRequestConditionApproved {
+			approvedConditions = append(approvedConditions, cond)
+		}
+
+		if cond.Type == cmapi.CertificateRequestConditionDenied {
+			deniedConditions = append(deniedConditions, cond)
+		}
+	}
```

Comment:
Could you use the `getCertificateRequestCondition` function here too?

Reply:
The problem is that that function will return the _first_ instance of either condition. That is fine since only one can be set, but this is the function that asserts this so we can't reuse it.